mentat/conversation.py

```diff
@@ -140,46 +138,56 @@ def get_messages(self) -> list[dict[str, str]]:
         else:
             parser = config.parser
             prompt = parser.get_system_prompt()
-            return [
-                {"role": MessageRole.System.value, "content": prompt}
-            ] + self._messages.copy()
+            prompt_message: ChatCompletionMessageParam = {
+                "role": "system",
+                "content": prompt,
+            }
+            return [prompt_message] + self._messages.copy()
 
     def clear_messages(self) -> None:
         """Clears the messages in the conversation"""
-        self._messages = list[dict[str, str]]()
+        self._messages = list[ChatCompletionMessageParam]()
 
     async def _stream_model_response(
         self,
-        messages: list[dict[str, str]],
+        messages: list[ChatCompletionMessageParam],
         loading_multiplier: float = 0.0,
     ):
-        start_time = default_timer()
         session_context = SESSION_CONTEXT.get()
         stream = session_context.stream
         config = session_context.config
         parser = config.parser
+        llm_api_handler = session_context.llm_api_handler
+
+        start_time = default_timer()
```

Comment:
Hmm... I would prefer to keep actual llm logic inside of llm_api_handler, but it could be useful; having the outside have to pass in an object that gets filled out seems awkward though. I think our current method is the simplest; let's just stick with it for now, but I can definitely see wrapping a timer generator around the response working in the future.

Reply:
The main reason I'm hesitating on adding a wrapped generator is we would just have to await the timer object that gets filled out when the generator is finished, which feels weird (since things could somehow go wrong and then we'd just get stuck waiting for the timer to fill out)